lib/dependencies/AMDDefineDependency.js

```diff
@@ -56,17 +56,29 @@ AMDDefineDependency.Template = class AMDDefineDependencyTemplate {
 				__WEBPACK_AMD_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_AMD_DEFINE_RESULT__))`
 			],
 			lf: [
-				"var XXX;",
-				"!(XXX = #.call(exports, __webpack_require__, exports, module))"
+				`var XXX,
+				 XXXmodule;
+				`,
+				`!(XXXmodule = { id: 1, exports: {}, loaded: false },
+					XXX = #.call(XXXmodule.exports, __webpack_require__, XXXmodule.exports, XXXmodule),
+					XXXmodule.loaded = true,
+					XXX === undefined && (XXX = XXXmodule.exports))`
 			],
 			lo: [
 				"var XXX;",
 				"!(XXX = #)"
 			],
 			lof: [
-				"var __WEBPACK_AMD_DEFINE_FACTORY__, XXX;",
-				`!(__WEBPACK_AMD_DEFINE_FACTORY__ = (#), XXX = (typeof __WEBPACK_AMD_DEFINE_FACTORY__ === 'function' ?
-				(__WEBPACK_AMD_DEFINE_FACTORY__.call(exports, __webpack_require__, exports, module)) : __WEBPACK_AMD_DEFINE_FACTORY__))`
+				`var XXX,
+				 XXXfactory,
+				 XXXmodule;
+				`,
+				`!(XXXfactory = (#),
+				 (XXXmodule = { id: 1, exports: {}, loaded: false }),
+				 XXX = (typeof XXXfactory === 'function' ?
+				 (XXXfactory.call(XXXmodule.exports, __webpack_require__, XXXmodule.exports, XXXmodule)) : XXXfactory),
+				 (XXXmodule.loaded = true),
+				 XXX === undefined && (XXX = XXXmodule.exports))`
```

Comment:
Hey, I added the id, removed new lines and added tests. Please, take a look once you will have time.